.github/workflows/main.yml

```diff
@@ -213,14 +213,23 @@ jobs:
         #load: true
 
   linux-build-test:
-    name: Ubuntu ${{ matrix.UbuntuVersion }} ${{ matrix.configuration }} build/test C++/C#
+    name: Ubuntu ${{ matrix.UbuntuVersion }} ${{ matrix.configuration }} build/test C++/C# (${{ matrix.cxx }})
     runs-on: ubuntu-latest
     needs: [prep-vars, docker-image-cached-build]
     timeout-minutes: 30
     strategy:
       matrix:
         UbuntuVersion: ${{ fromJson(needs.prep-vars.outputs.UbuntuVersionMatrix) }}
         configuration: [Debug, Release]
+        cxx: [clang++-10, g++]
+        include:
+          - cxx: clang++-10
+            cc: clang-10
+          - cxx: g++
+            cc: gcc
+        exclude:
+          - UbuntuVersion: 16.04
+            cxx: g++
```

Comment:
g++ on Ubuntu 16.04 is quite old and fails with some additional errors.
Rather than debug them, for the moment, I simply disabled it.

Reply:
@grlap says we can actually enable this too by fixing `constexpr`

Reply:
@grlap I pulled some of your changes in and added a few others to fix some `std::array` initialization syntax that `gcc` on 16.04 was complaining about.